src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala

```diff
@@ -58,13 +75,20 @@ object DataTypes {
           attributesOpt.flatMap { attributes =>
             attributes
               .find(_.name == aggregation.field)
-              .map(attribute => canBeAggregated(attribute.dataType) -> aggregation.field)
+              .map(attribute => canBeAggregated(attribute.dataType, aggregation.function) -> aggregation.field)
           }
         }
       }
       .map(_.flatten.collect { case (false, fieldName) => InvalidAggregationFieldForType(fieldName) }.toList)
   }
 
+  /** Method checks if type can be aggregated */
+  def canBeAggregated(dataType: DataType, aggregationType: AggregationType): Boolean = {
+    if (aggregationType != AggregationType.count) {
+      Set(DataType.Decimal, DataType.Int, DataType.LargeInt, DataType.DateTime)(dataType)
```

Comment:
_Tag: performance_
I would try to make this result a constant, since it will be rebuilt each time the method is called.

Alternatively the method could be somehow converted to a function value?

Reply:
yes, I started with that, but I didn't like how it looked :P I'll change that ;)

Reply:
It's not that critical, anyway